specification/azsadmin/resource-manager/storage/Microsoft.Storage.Admin/preview/2019-08-08/storageaccounts.json

```diff
@@ -0,0 +1,593 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2019-08-08-preview",
+    "title": "StorageManagementClient",
+    "description": "The Admin Storage Management Client."
+  },
+  "host": "adminmanagement.local.azurestack.external",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.Storage.Admin/storageaccounts": {
+      "get": {
+        "x-ms-examples": {
+          "Returns a list of storage accounts.": {
+            "$ref": "./examples/StorageAccounts/List.json"
+          }
+        },
+        "tags": [
+          "StorageAccounts"
+        ],
+        "operationId": "StorageAccounts_List",
+        "description": "Returns a list of storage accounts.",
+        "parameters": [
+          {
+            "$ref": "storageaccounts.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "storageaccounts.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "storageaccounts.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "storageaccounts.json#/parameters/FilterParameter"
+          },
+          {
+            "$ref": "storageaccounts.json#/parameters/SummaryParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- The list of storage accounts has been returned.",
+            "schema": {
+              "$ref": "storageaccounts.json#/definitions/StorageAccountList"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.Storage.Admin/storageaccounts/{accountId}": {
+      "get": {
+        "x-ms-examples": {
+          "Returns the requested storage account.": {
+            "$ref": "./examples/StorageAccounts/Get.json"
+          }
+        },
+        "tags": [
+          "StorageAccounts"
+        ],
+        "operationId": "StorageAccounts_Get",
+        "description": "Returns the requested storage account.",
+        "parameters": [
+          {
+            "$ref": "storageaccounts.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "storageaccounts.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "storageaccounts.json#/parameters/AccountIdParameter"
+          },
+          {
+            "$ref": "storageaccounts.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- The storage account has been returned.",
+            "schema": {
+              "$ref": "storageaccounts.json#/definitions/StorageAccount"
+            }
+          },
+          "404": {
```

Comment:
we only list success codes here, all others should be covered in the 'default' cloud error pattern.

 "default": {
            "description": "Error response",
            "schema": {
              "$ref": "#/definitions/CloudError"
            }
          }

All apis should include this.

Reply:
This spec is for azurestack admin scenario, which may not obey the tenant error rule, but I will extract an error base class for all the admin requests.